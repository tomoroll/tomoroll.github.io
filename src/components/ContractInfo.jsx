import React from 'react';
import PropTypes from 'prop-types';
import Address from './Address';

function ContractInfo({
  accountAddress, accountBalance, contractAddress, contractBalance, network,
}) {
  const contractAddr = <Address network={network} address={contractAddress} />;
  const contractBalanceBlock = (
    <div className="col-12 d-none d-sm-block">
      <i className="far fa-file-code" />
      &nbsp;Contract&nbsp;(
      {contractBalance.toFixed(2)}
      &nbsp;ETH)
    </div>
  );
  const contractAddressBlock = (
    <div className="col-12 d-none d-sm-block">
      {contractAddr}
    </div>
  );
  const accountAddr = (accountAddress !== null)
    ? <Address network={network} address={accountAddress} />
    : <span>Not connected, please login to MetaMask</span>;
  const accountBalanceBlock = (
    <div className="col-lg-12">
      <i className="far fa-user" />
      &nbsp;Account&nbsp;(
      {accountBalance.toFixed(2)}
      &nbsp;ETH)
    </div>
  );
  const accountAddressBlock = (
    <div className="col-12">
      {accountAddr}
    </div>
  );
  return (
    <div className="row">
      {accountBalanceBlock}
      {accountAddressBlock}
      {contractBalanceBlock}
      {contractAddressBlock}
    </div>
  );
}
ContractInfo.propTypes = {
  accountAddress: PropTypes.string,
  accountBalance: PropTypes.number.isRequired,
  contractAddress: PropTypes.string.isRequired,
  contractBalance: PropTypes.number.isRequired,
  network: PropTypes.number.isRequired,
};
ContractInfo.defaultProps = {
  accountAddress: null,
};

export default ContractInfo;
