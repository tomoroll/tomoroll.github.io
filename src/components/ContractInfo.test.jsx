import React from 'react';
import renderer from 'react-test-renderer';
import ContractInfo from './ContractInfo';

it('renders correctly', () => {
  const tree = renderer
    .create(<ContractInfo accountAddress="0x0123" accountBalance={1.123} contractAddress="0x3210" contractBalance={123} network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});


it('no account address', () => {
  const tree = renderer
    .create(<ContractInfo contractAddress="0x3210" contractBalance={123} accountBalance={0} network={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
