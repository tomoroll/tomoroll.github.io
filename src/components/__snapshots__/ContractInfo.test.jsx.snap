// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no account address 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12"
  >
    <i
      className="far fa-user"
    />
     Account (
    0.00
     ETH)
  </div>
  <div
    className="col-12"
  >
    <span>
      Not connected, please login to MetaMask
    </span>
  </div>
  <div
    className="col-12 d-none d-sm-block"
  >
    <i
      className="far fa-file-code"
    />
     Contract (
    123.00
     ETH)
  </div>
  <div
    className="col-12 d-none d-sm-block"
  >
    <a
      className="text-monospace"
      href="https://ropsten.etherscan.io/address/0x3210"
    >
      0x3210
    </a>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-12"
  >
    <i
      className="far fa-user"
    />
     Account (
    1.12
     ETH)
  </div>
  <div
    className="col-12"
  >
    <a
      className="text-monospace"
      href="https://ropsten.etherscan.io/address/0x0123"
    >
      0x0123
    </a>
  </div>
  <div
    className="col-12 d-none d-sm-block"
  >
    <i
      className="far fa-file-code"
    />
     Contract (
    123.00
     ETH)
  </div>
  <div
    className="col-12 d-none d-sm-block"
  >
    <a
      className="text-monospace"
      href="https://ropsten.etherscan.io/address/0x3210"
    >
      0x3210
    </a>
  </div>
</div>
`;
